toolkit/smart-contracts/offchain/src/governed_map/mod.rs

```diff
@@ -155,5 +157,151 @@ fn insert_key_value_tx(
 	Ok(tx_builder.balance_update_and_build(ctx)?.remove_native_script_witnesses())
 }
 
+pub async fn run_remove<
+	C: QueryLedgerState + QueryNetwork + Transactions + QueryUtxoByUtxoId,
+	A: AwaitTx,
+>(
+	genesis_utxo: UtxoId,
+	key: String,
+	payment_signing_key: &CardanoPaymentSigningKey,
+	ogmios_client: &C,
+	await_tx: &A,
+) -> anyhow::Result<Option<MultiSigSmartContractResult>> {
+	let ctx = TransactionContext::for_payment_key(payment_signing_key, ogmios_client).await?;
+	let (validator, policy) = crate::scripts_data::governed_map_scripts(genesis_utxo, ctx.network)?;
+	let validator_address = validator.address_bech32(ctx.network)?;
+	let validator_utxos = ogmios_client.query_utxos(&[validator_address]).await?;
+
+	let utxos_for_key = get_utxos_for_key(validator_utxos, key.clone(), policy.policy_id())?;
+	let tx_hash_opt = match utxos_for_key.len() {
+		0 => {
+			log::info!("There is no value stored for key '{}'. Skipping remove.", key);
+			None
+		},
+		_ => Some(
+			remove(&validator, &policy, &utxos_for_key, ctx, genesis_utxo, ogmios_client).await?,
+		),
+	};
+	if let Some(TransactionSubmitted(tx_hash)) = tx_hash_opt {
+		await_tx.await_tx_output(ogmios_client, UtxoId::new(tx_hash.0, 0)).await?;
+	}
+	Ok(tx_hash_opt)
+}
+
+fn get_utxos_for_key(
+	utxos: Vec<OgmiosUtxo>,
+	key: String,
+	token: PolicyId,
+) -> anyhow::Result<Vec<OgmiosUtxo>> {
+	let mut utxos_for_key = Vec::new();
+	for utxo in utxos {
+		if let Some(datum) = utxo.clone().datum {
```

Comment:
In this PR: https://github.com/input-output-hk/partner-chains/pull/674 you've approved different behavior. We should make up our minds.

Also `get_current_value` should use `get_utxos_for_key`